### ISSUE TITLE
Enforce HSTS - Strict Transport Security

### ISSUE THREAD

**Issue by sfllaw** [sourcegraph/issues#81]
- **Issue type:** Bug report
- **Sourcegraph version:** 2.7.6 (15680_2018-05-18_2d07482)
- **Platform:** N/A

#### Steps to reproduce:

1. Set `"httpToHttpsRedirect": "load-balanced"` in the site configuration.
2. Run the following HTTPS request:
```console
sfllaw@host:~$ http https://sourcegraph.example.com/
HTTP/1.1 302 Found
Access-Control-Allow-Credentials: true
Access-Control-Allow-Origin: https://github.example.com
Cache-Control: no-cache, max-age=0
Connection: keep-alive
Content-Length: 48
Content-Type: text/html; charset=utf-8
Date: Fri, 18 May 2018 04:03:04 GMT
Location: /.auth/login?redirect=%2F
Vary: Cookie
X-Content-Type-Options: nosniff
X-Frame-Options: DENY
X-Trace: #tracer-not-enabled
X-Xss-Protection: 1; mode=block

<a href="/.auth/login?redirect=%2F">Found</a>.
```

#### Expected:
`Strict-Transport-Security: max-age=31536000; includeSubDomains`

#### Actual:
HTTP Strict Transport Security is not enabled, so browsers can fall back to HTTP, when they don't need to.

See https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Strict-Transport-Security

**Comment by nicksnyder** [comment]
There is a separate site config option `"httpStrictTransportSecurity"` which you can set to be whatever you want the header to be. `"httpStrictTransportSecurity": "max-age=31536000; includeSubDomains"`

**Comment by sfllaw** [comment]
@nicksnyder I see. The documentation isn't updated on your website, so I didn't discover this option.

Please make sure that `httpToHttpsRedirect`'s documentation mentions this as good practice.

**Comment by sfllaw** [comment]
@nicksnyder I see. The documentation isn't updated on your website, so I didn't discover this option. Confirmed that this option exists and works correctly in 2.7.6 (15680_2018-05-18_2d07482).

Please make sure that `httpToHttpsRedirect`'s documentation mentions this as good practice. You may even want to have a sensible default that admins can override to `""` if they want to suppress the header.

**Comment by nicksnyder** [comment]
Yeah we just added this option yesterday and the doc site hasn't been updated yet :)

I agree that it would be ideal to have the header set by default.

**Comment by nicksnyder** [comment]
I just pushed a commit to master that makes HSTS on by default.
